packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -525,6 +533,20 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
     log.debug(`<< Payment - Peer ${peerAccount} Balance: ${balance} Downloaded ${downloaded}`);
   }
 
+  /** Utility function. Checks if the budget has enough receive capacity to accept a new channel. Pauses the torrent if it does not. **/
+  private checkUploadBudget(budget, torrent) {
+    if (budget?.budgets?.length === 1) {
+      const receiveCapacity = bigNumberify(budget.budgets[0].availableReceiveCapacity);
+      const uploadSize = WEI_PER_BYTE.mul(torrent.length).toHexString();
+      if (receiveCapacity.lt(uploadSize)) {
+        log.warn(
+          `Receive capacity (${receiveCapacity.toHexString()}) is less than file cost (${uploadSize}). Pausing torrent.`
+        );
+        torrent.pause();
```

Comment:
I'm not sure if this does what I want it to. I could `destroy` the torrent but that would cause existing connections to fail.